### DISCUSSION TITLE
PDF Viewer for attachment is failing due to expected xref at start of table

### DISCUSSION THREAD

**Discussion by michaeljeffreycollins** [nextgenhealthcare/connect#6137] [Q&A] [answered]
This question has been asked in various forms over the years on both forums. But I cannot use a file writer or the question remained unanswered. I have a process that pulls an embedded pdf from OBX.5 and stores it as an attachment. We have tested and used this process for years so we know that it is functioning properly. 

Sometimes we do have an error whether it is network, data, or corruption. We then turn to a number of tools to assist us. One of those tools is the pdf viewer so we can see the PDF.  Rarely the attachment viewer fails due to xref expected at start of table. I would say it works fine at least 85% of the time. 

The error message in question makes little sense. But the xref is only required as of pdf reference up to version 3 which published in 2001. The next revision made it optional. So, the attachment view is trying to render a pdf using a 23 year old reference. I fully suspect my PDF is perfectly fine, but I cannot view it. We are not using a long term network archive where I can just drop a file with a filewriter. Instead this HL7 message is sending the pdf to our long term document storage solution. 

Is it possible to switch the PDF reference version on the viewer? Can I use a different viewer? Is it possible to improve the pdf viewer functionality? Is there a completely different way to view the contents of the pdf? 

Today I have a series of these messages failing and we cannot tell if the PDF is corrupted or not. I am turning to the community for suggestions and workarounds. Thank you

**Comment by pacmano1** [comment, marked as answer]
Fair ask..  the viewer has failed so long on assorted documents I resigned myself to just save it to disk and open it in some other tool.

I would open a bug or feature request in the "issues" section.

**Reply by michaeljeffreycollins** [reply]
You make a super good point. I will replicate this over to the issues section.

**Reply by tonygermano** [reply]
https://github.com/nextgenhealthcare/connect/issues/6138
